internal/provider/common/previews.go

```diff
@@ -0,0 +1,41 @@
+package common
+
+import (
+	"context"
+
+	"terraform-provider-render/internal/client"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+)
+
+// PreviewsToPreviewsObject generates a types.Object because `previews` is Generated and Optional, so Terraform needs
+// the types.Object so it knows if it is Unknown versus Null.
```

Comment:
I'm a little confused at what this means here. 
Is a nil value for previews an indicator that the type could be Unknown? What does an Unknown type here mean in this case?

Reply:
Unknown means the Terraform convergence process has not yet determined what the correct value is, and requires a read or a return from a create or update to say that it knows the value. Known and Unknown are states relevant to the Terraform lifecycle.